eZ/Bundle/EzPublishCoreBundle/EventListener/RequestEventListener.php

```diff
@@ -82,6 +82,9 @@ public function onKernelRequestForward(GetResponseEvent $event)
                     $request->getContent()
                 );
                 $forwardRequest->attributes->add($request->attributes->all());
+                if ($request->headers->has('X-User-Hash')) {
+                    $forwardRequest->headers->set('X-User-Hash', $request->headers->get('X-User-Hash'));
+                }
```

Comment:
this should not be hardcoded but use an injected user_hash_header

maybe instead of doing this, the listener in FOSHttpCache should detect forwarded requests and decide the 'Vary' based on the master request. i guess the forward keeps the session created from cookie/auth header intact? is it the same with sub requests or do those keep the original headers as varnish sent them?